sdk/identity/azure-identity/src/workload_identity_credential.cpp

```diff
@@ -100,11 +100,21 @@ WorkloadIdentityCredential::WorkloadIdentityCredential(
 
 WorkloadIdentityCredential::~WorkloadIdentityCredential() = default;
 
+std::string WorkloadIdentityCredential::GetAssertion(Context const&)
```

Comment:
This does not modify the internal state of the class, it should be made `const`
```suggestion
std::string WorkloadIdentityCredential::GetAssertion(Context const&) const
```

Reply:
FWIW, I think that all methods that *can* be const *should* be const. There is basically no harm to making them const, and it allows the compiler to do a much better job of optimizing (for instance, it can cache values retrieved from the class across const method calls because it knows that the values wouldn't have been modified by the method call).

Reply:
Removing it could in theory be breaking b/c a const object can't call a non-const member function. One could continue to support both cases by overloading the method as its a legal overload to have both
`int Bar() const;` and `int Bar();`

Reply:
> its a legal overload to have both `int Bar() const;` and `int Bar();`

TIL
```cpp
#include <iostream>

class Foo
{
public:
    int Bar() const
    {
        return 1;
    }
    int Bar()
    {
        return 0;
    }
};

int main() {
    auto non_const = Foo();
    std::cout << non_const.Bar() << std::endl; // 0
    
    const auto const_foo = Foo();
    std::cout << const_foo.Bar() << std::endl; // 1

    return 0;
}
```